RN 1.2.0 (#165)
percona/everest-doc: docs/release-notes/Percona-Everest-1.2.0-(2024-09-09).md

- With the upgrade to Percona Everest version 1.2.0, the transition from global scope to the designated namespaces for these resources is an important change in the way access control is managed. This improves security as the resources are only accessible within their designated namespaces. The database clusters can only use `monitoring-instances` and `backup-storages` located within the same namespace as the cluster.

- When upgrading to 1.2.0 using the CLI command `everestctl upgrade`, all your existing `backup-storages` and `monitoring-instances` will be automatically migrated to the namespaces specified in their `.spec.allowedNamespaces` fields.

    !!! note
        After the upgrade to Percona Everest 1.2.0, you will only be able to access these resources through the new API endpoints.

    Check out our [documentation](docs.percona.com/everest//administer/api_rbac.html#changes-in-the-percona-everest-apis) for in-depth details on the Breaking API changes.

### Improved multiple operator upgrades

> Starting with Percona Everest 1.2.0, we have the capability to upgrade all the database operators concurrently with their components across any namespace using our intuitive UI with just a single click.

Reviewer: I think this is slightly misleading in the sense that this option is not just a new capability, but the only option available. Earlier we would allow (or at least make it seem like) upgrading just a single operator, but that is no longer an option now because of a technical limitation

Author: ok. Lets discuss with @recharte.

Author: @mayankshah1607 updated pls check

- [EVEREST-1142](https://perconadev.atlassian.net/browse/EVEREST-1142): We have now added a [new command](../administer/rbac.md#validate-your-rbac-policy) for validating your RBAC policy to ensure that your RBAC policies are working as expected.

- [EVEREST-1240](https://perconadev.atlassian.net/browse/EVEREST-1240): We have added support for PostgreSQL operator version 2.4.1. 

- [EVEREST-1298](https://perconadev.atlassian.net/browse/EVEREST-1298): We have added support for MySQL operator version 1.15.0.

- [EVEREST-1303](https://perconadev.atlassian.net/browse/EVEREST-1303): Starting with Percona Everest 1.2.0, you can enable sharding for MongoDB when creating a database cluster. By leveraging sharding in Percona Everest, you can enhance the performance and scalability of your database clusters.

- [EVEREST-981](https://perconadev.atlassian.net/browse/EVEREST-981): Configure everest to run behind a proxy

## Improvements

> - [EVEREST-1165](https://perconadev.atlassian.net/browse/EVEREST-1165)- We now have the capability to upgrade all database operators simultaneously with their components across any namespace using our user-friendly interface with just a single click.

Reviewer: Same above point about upgrades applies here

Author: @mayankshah1607 Fixed. Pls check

- [EVEREST-1369](https://perconadev.atlassian.net/browse/EVEREST-1369): We've updated the documentation for the section on turning off telemetry.

## Bugs

- [EVEREST-768](https://perconadev.atlassian.net/browse/EVEREST-768): We've resolved the issue with the PostgreSQL pods not restarting when the database was restarted.

- [EVEREST-1232](https://perconadev.atlassian.net/browse/EVEREST-1232): The backups and restore pages have been updated to show consistent date formats .

- [EVEREST-1253](https://perconadev.atlassian.net/browse/EVEREST-1253): The **Delete** option in the backup menu is properly now disabled if the backup is in the **Deleting** status.

- [EVEREST-1273](https://perconadev.atlassian.net/browse/EVEREST-1273) Backend sends an empty response if the backup storage url is edited to an http address

- [EVEREST-1279](https://perconadev.atlassian.net/browse/EVEREST-1279): The CPU value displayed on the Resources page and the Database Summary panel is now consistent.

- [EVEREST-1286](https://perconadev.atlassian.net/browse/EVEREST-1286): When editing a backup schedule, the option to change the backup name is now disabled.

> - [EVEREST-1287](https://perconadev.atlassian.net/browse/EVEREST-1287) No ability to change antiAffinityTopologyKey or other affinity settings

Reviewer: ```suggestion
- [EVEREST-1287](https://perconadev.atlassian.net/browse/EVEREST-1287) Updated the default affinity settings for the database cluster pods. A database restart is required in order to apply these new settings.
```

- [EVEREST-1315](https://perconadev.atlassian.net/browse/EVEREST-1315): Despite Percona Everest showing a successful upgrade message, the upgrade actually didn't go as planned. The issue has been resolved now.

- [EVEREST-1323](https://perconadev.atlassian.net/browse/EVEREST-1323):We've resolved the issue that was causing an error to appear on the **Components** page after creating databases, especially when the status was either **initializing** or **up**.

> - [EVEREST-1354](https://perconadev.atlassian.net/browse/EVEREST-1354): We've resolved an issue related to the incorrect display of CPU and memory resources in the PXC CRD.

Reviewer: ```suggestion
- [EVEREST-1354](https://perconadev.atlassian.net/browse/EVEREST-1354): We've resolved an issue related to an incorrect amount of CPU and memory resources being allocated for PXC clusters. A database restart is required in order to apply these new settings.
```

Author: Instead of this we can write @mayankshah1607 :

We've resolved an issue related to an incorrect allocation of CPU and memory resources for PXC clusters. To apply these new settings, a database restart is required.

Author: Fixed

# What's new in Percona Everest 1.2.0

To begin your journey with Percona Everest, check out the [Quickstart Guide for Percona Everest](../quickstart-guide/quick-install.md).

> 

Reviewer 2: We need a big warning sign at the beginning letting users know about the backward incompatible changes.

Author: Yeah.

Author: Done.

- With the upgrade to Percona Everest version 1.2.0, the transition from global scope to the designated namespaces for these resources is an important change in the way access control is managed. This improves security as the resources are only accessible within their designated namespaces. The database clusters can only use `monitoring-instances` and `backup-storages` located within the same namespace as the cluster.

- When upgrading to 1.2.0 using the CLI command `everestctl upgrade`, all your existing `backup-storages` and `monitoring-instances` will be automatically migrated to the namespaces specified in their `.spec.allowedNamespaces` fields.

    !!! note
        After the upgrade to Percona Everest 1.2.0, you will only be able to access these resources through the new API endpoints.

    Check out our [documentation](docs.percona.com/everest//administer/api_rbac.html#changes-in-the-percona-everest-apis) for in-depth details on the Breaking API changes.

> ### Exploring the benefits of MongoDB sharding in Percona Everest
> 
> [Sharding  :octicons-link-external-16:](https://docs.mongodb.com/manual/reference/glossary/#term-sharding) is used for horizontal database scaling. It distributes a database horizontally across multiple nodes or servers, known as **shards**. Each shard manages a portion of the data, forming a sharded cluster, which enables MongoDB to handle large datasets and high user concurrency effectively.
> 
> MongoDB sharding has several key components:
> 
> - shard: Each shard has a subset of the data.
> - mongos: The query router directs the client queries to the proper shard(s).
> 
>     !!! info "Important"
>         With the current Percona Everest release, the number of routers (mongos) is, by default, set to the number of nodes in your replica set. However, in our upcoming releases, you can customize the number of routers and set resource limits for them.
> 
> - config servers: The configuration servers store the cluster's metadata and configuration settings.
> 
> #### Enable sharding in Percona Everest
> 
> To enable sharding:
> {.power-number}
> 
> 1. On the **Create Database** wizard, select **MongoDB** database and turn on the **Sharded Cluster** toggle.
> 
>     ![!image](../images/enable_sharding.png)
> 
> 
> 2. To check if sharding is enabled, go to the database view page and click on the specific database. Then, check the **Resources** panel to see if sharding is enabled.
> 
> 
>     ![!image](../images/sharding_status.png)

Reviewer 2: Moved to 1.3
```suggestion
```

- [EVEREST-1298](https://perconadev.atlassian.net/browse/EVEREST-1298): We have added support for MySQL operator version 1.15.0.

- [EVEREST-1035](https://perconadev.atlassian.net/browse/EVEREST-1035): We've now included **Retention copies** for PostgreSQL as well when setting up backup schedules.

## Improvements

- [EVEREST-1165](https://perconadev.atlassian.net/browse/EVEREST-1165)- Due to limitations with the [Operator Lifecycle Manager (OLM)](https://olm.operatorframework.io/), it is now required to upgrade all database operators concurrently with their components across any namespace.

- [EVEREST-1212](https://perconadev.atlassian.net/browse/EVEREST-1212) - Starting with Percona Everest 1.2.0, you can now directly edit the monitoring endpoint from the database overview page, instead of having to use the Edit database wizard.

- [EVEREST-1230](https://perconadev.atlassian.net/browse/EVEREST-1230): We've updated the **Resources** panel on the **Database overview** page to be independent instead of part of the **DB Details** panel and improved the overall look and feel of this page.

> - [EVEREST-1369](https://perconadev.atlassian.net/browse/EVEREST-1369): We've updated the documentation for the section on turning off telemetry.
> 
> 

Reviewer 2: Let's keep this out of the RNs
```suggestion
```

## The latest in APIs: What’s new and what’s deprecated

=== "Newly added API endpoints"

    Check out the new API endpoints we've added in Percona Everest 1.2.0:

    |**No**|**API endpoints**|**Method**|
    |---------|---------------------|---------|
    |**1.**|`/namespaces/{namespace}/monitoring-instances`|a.`GET`</br> b.`POST`|
    |**2.**|`/namespaces/{namespace}/monitoring-instances/{name}`|a.`GET`</br>b. `PATCH`</br>c.`DELETE`|
    |**3.**|`/namespaces/{namespace}/backup-storages`|a.`GET`</br> b. `POST`|
    |**4.**|`/namespaces/{namespace}/backup-storages/{name}`|a.`GET`</br> b.`POST`|
    |**5.**|``/permissions``|a.`GET`|

=== "Deprecated API endpoints"

    This is the list of the API endpoints deprecated from Percona Everest:

    |**No**|**API endpoints**|**Method**|
    |---------|---------------------|---------|

>     |**1.** |**Endpoints deprecated from Percona Everest v1.1.0:**|

Reviewer 2: We already discussed this offline but let's explicitly say that these endpoint that were deprecated in 1.1.0 have now been removed from 1.2.0

    |**a.**|`/namespaces/{namespace}/database-engines/{name}/operator-version/preflight`|1.`GET`|
    |**b.**|`/namespaces/{namespace}/database-engines/{name}/operator-version`|1.`GET`</br>2.`PUT`|
    |**2.** |**Endpoints deprecated from Percona Everest v1.2.0:**|
    |**c.**|`/monitoring-instances`|1.`GET`</br> 2.`POST`|
    |**d.**|`/monitoring-instances/{name}`|1.`GET`</br>2. `PATCH`</br>3.`DELETE`|
    |**e.**|`/backup-storages`|1.`GET`</br> 2.`POST`|
    |**f.**|`/backup-storages/{name}`|1.`GET`</br>2. `PATCH`</br>3.`DELETE`|

## Bugs

- [EVEREST-768](https://perconadev.atlassian.net/browse/EVEREST-768): The PostgreSQL pods now demonstrate the intended behavior by not getting stuck, as they automatically restart when the database is restarted.

- [EVEREST-1232](https://perconadev.atlassian.net/browse/EVEREST-1232): The backups and restore pages have been updated to show consistent date formats.

- [EVEREST-1253](https://perconadev.atlassian.net/browse/EVEREST-1253): The **Delete** option in the backup menu is properly now disabled if the backup is in the **Deleting** status.

> - [EVEREST-1273](https://perconadev.atlassian.net/browse/EVEREST-1273) Backend sends an empty response if the backup storage URL is edited to an HTTP address.

Reviewer 2: The fix is that now we show an error message if the user edits the backup storage URL to an invalid one.

- [EVEREST-1315](https://perconadev.atlassian.net/browse/EVEREST-1315): Despite Percona Everest showing a successful upgrade message, the upgrade actually didn't go as planned. The issue has been resolved now.

- [EVEREST-1323](https://perconadev.atlassian.net/browse/EVEREST-1323):We've resolved the issue that was causing an error to appear on the **Components** page after creating databases, especially when the status was either **initializing** or **up**.

- [EVEREST-1354](https://perconadev.atlassian.net/browse/EVEREST-1354): We've resolved an issue related to an incorrect allocation of CPU and memory resources for PXC clusters. To apply these new settings, a database restart is required.

- [EVEREST-1371](https://perconadev.atlassian.net/browse/EVEREST-1371): We've addressed a couple of issues pertaining to the **Monitoring** page, one of them being the monitoring endpoint displaying the **username** instead of the **endpoint** name.

- [EVEREST-1372](https://perconadev.atlassian.net/browse/EVEREST-1372): We’ve resolved the issue that prevented us from deleting the monitoring endpoint that is not currently in use.

> - [EVEREST-1375](https://perconadev.atlassian.net/browse/EVEREST-1375): The upgrade to PXC operator 1.15.0 caused a restart of the databases. We have resolved the issue now.
> 

Reviewer 2: This issue was only in one of the dev builds
```suggestion
```

> - [EVEREST-1407](https://perconadev.atlassian.net/browse/EVEREST-1407) - After upgrading to Percona Everest 1.2.0, it was observed that a user who had not been included in the RBAC configuration and lacked proper permissions, was able to access specific information on Percona Everest. We have resolved the issue now.
> 
> 

Reviewer 2: Dev build issue
```suggestion
```

- [EVEREST-1427](https://perconadev.atlassian.net/browse/EVEREST-1427) - 
PostgreSQL no longer gets stuck in an unknown state despite having duplicate backup storage in different namespaces.

> - [EVEREST-1465](https://perconadev.atlassian.net/browse/EVEREST-1465) - Fixed an issue where a restore to the same and new database could be performed, even if the user lacked the necessary database-cluster-credentials permissions.
> 
> - [EVEREST-1466](https://perconadev.atlassian.net/browse/EVEREST-1466) - Restoring to the same MySQL database was stuck despite the user having the necessary permissions to restore the databases. We have resolved the issue now.
> 
> - [EVEREST-1472](https://perconadev.atlassian.net/browse/EVEREST-1472) - The option to **Add backup storage **is no longer visible when editing a database cluster if the user does not have the necessary permissions.
> 
> - [EVEREST-1473](https://perconadev.atlassian.net/browse/EVEREST-1473) - Resolved an issue where the backup API returned information about the backup storage location to a user without permission for backup storage locations.
> 
> - [EVEREST-1477](https://perconadev.atlassian.net/browse/EVEREST-1477) - Resolved an issue where the database cluster API returned information about the storage name and schedules to a user without permission for backups and storage.
> 
> - [EVEREST-1486](https://perconadev.atlassian.net/browse/EVEREST-1486) - You can now view the database credentials for databases in a namespace on the **database** page as well as the **database overview** page if they have the necessary permissions, as intended.
> 

Reviewer 2: All dev build bugs
```suggestion
```
